plugins/modules/rds_instance.py

```diff
@@ -938,7 +956,9 @@ def get_changing_options_with_inconsistent_keys(modify_params, instance, purge_c
         # TODO: allow other purge_option module parameters rather than just checking for things to add
         if isinstance(current_option, list):
             if isinstance(desired_option, list):
-                if set(desired_option) <= set(current_option):
+                if set(desired_option) < set(current_option):
```

Comment:
The `continue` below used to happen for `<=`. You could maybe consolidate the special case into a single conditional and add an elif for the generic case.

```python
if (
    set(dedirect_option) < set(current_option) and
    option in ('DBSecurityGroups', 'VpcSecurityGroupIds',) and
    purge_security_groups
):
    ...
elif set(desired_option) <= set(current_option):
    ...

Reply:
@ s-hertel Thank you so much for reviewing it. In fact, I wasn't sure whether to return all security groups or only those with added/active status. Thanks for explaining and it makes sense that the others have the freedom to decide. I applied this change for the conditional statement.